### PR TITLE
feat: sp adds http probe to improve service stability

### PR DESCRIPTION
**Pull request by sysvm**
### Description

SP adds HTTP probe to improve service stability.

### Rationale

For being noticed in time when a service is deployed in Kubernetes what kinds of status.

### Example

```yaml
livenessProbe:
  failureThreshold: 8
  httpGet:
    path: /-/healthy
    port: 24369
    scheme: HTTP
  periodSeconds: 30
  name: sp
ports:
- containerPort: 24369
  name: http
readinessProbe:
  failureThreshold: 20
  httpGet:
    path: /-/ready
    port: 24369
    scheme: HTTP
  periodSeconds: 5
```
### Changes

Notable changes: 
* N/A

### CODE REVIEW

File: pkg/probe/http.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+			return
+		}
+		if _, err := io.WriteString(w, "OK"); err != nil {
+			log.Errorw("failed to write probe response", "error", err)
+		}
+	}
+}
+
+// IsReady returns true if component is ready.
+func (p *HTTPProbe) IsReady() bool {
```
Comment (will-2012) [inline pkg/probe/http.go:47]: checkReady?

File: pkg/probe/http.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	value := p.ready.Load()
+	return value > 0
+}
+
+// isHealthy returns true if component is healthy.
+func (p *HTTPProbe) isHealthy() bool {
```
Comment (will-2012) [inline pkg/probe/http.go:53]: checkHealthy

File: pkg/probe/http.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	value := p.healthy.Load()
+	return value > 0
+}
+
+// Ready sets components status to ready.
+func (p *HTTPProbe) Ready() {
```
Comment (will-2012) [inline pkg/probe/http.go:59]: SetReady?

File: base/gfspapp/app.go [hunk 4]
```diff
@@ -50,6 +52,8 @@ type GfSpBaseApp struct {
 	uploader      module.Uploader
 	metrics       module.Modular
 	pprof         module.Modular
+	probeSvr      module.Modular
+	httpProbe     coreprober.Prober
```
Comment (will-2012) [inline base/gfspapp/app.go:56]: Remove this `httpProbe`?

Comment (sysvm) [reply base/gfspapp/app.go:56]: why?

File: pkg/probe/http.go (hunk 1 above)
Comment (sysvm) [reply pkg/probe/http.go:47]: Are there any differences?

**Review by will-2012 — APPROVED** [review]

**Pull request merged**